### PR TITLE
chore: document state machine engine

### PR DESCRIPTION
**Pull request by div5yesh**
*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [ ] Added Unit Tests
- [ ] Tested on frontend (Add Screenshots)
- [ ] Successful logs (Attach them to the PR)

*Documentation update required?*
- [x] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt [hunk 1]
```diff
@@ -101,17 +136,30 @@ internal open class StateMachine<StateType : State, EnvironmentType : Environmen
...
+     * Unregister a listener.
+     * @param token token of the listener to remove
+     */
     private fun removeSubscription(token: StateChangeListenerToken) {
         pendingCancellations.remove(token)
         subscribers.remove(token)
     }
 
+    /**
+     * Sends `event` to the StateMachine for resolution, and applies any effects and new states returned from the resolution.
```
Comment (tjleing) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt:149]: Can you do a pass for verb tense here?  Like how this function `sends` an event but the next one below `notif(ies)` all the listeners, etc.  Do we have a style guide for Javadoc-type comments?

**Review by tjleing — APPROVED** [review]

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt [hunk 2]
```diff
@@ -28,9 +28,15 @@ typealias StateChangeListenerToken = UUID
...
- * Resolvers of type State.
+ * Models, mutates and processes effects of a system as a finite state automaton. It consists of:
+ * State - which represents the current state of the system
+ * Resolver - a mechanism for mutating state in response to events and returning side effects called Actions
+ * Listener - which accepts and enqueues incoming events
+ * StateChangedListeners - which are notified whenever the state changes
+ * EffectExecutor - which resolves and executes side Effects/Actions
  * @implements EventDispatcher
+ * @param resolver responsible for mutating state based on incoming events
+ * @param environment holds system specific environment info accessible to Effects/Actions
```
Comment (eeatonaws) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt:39]: Do we need comments describing the other parameters (executor, concurrentQueue, initialState) as well?

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt [hunk 3]
```diff
@@ -101,17 +136,30 @@ internal open class StateMachine<StateType : State, EnvironmentType : Environmen
...
+     */
     private fun removeSubscription(token: StateChangeListenerToken) {
         pendingCancellations.remove(token)
         subscribers.remove(token)
     }
 
+    /**
+     * Sends `event` to the StateMachine for resolution, and applies any effects and new states returned from the resolution.
+     */
     override fun send(event: StateMachineEvent) {
```
Comment (eeatonaws) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt:151]: Missing `@param` comment for `event`.

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachineResolver.kt [hunk 4]
```diff
@@ -18,6 +18,26 @@ package com.amplifyframework.statemachine
 import java.util.logging.Level
 import java.util.logging.Logger
 
+/**
+ * Executes a state transition when a precondition is fulfilled or when an event is received.
+ *
+ * StateMachineEvent are resolved as follows:
```
Comment (eeatonaws) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachineResolver.kt:24]:
```suggestion
 * StateMachineEvents are resolved as follows:
```

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt (hunk 1 above)
Comment (div5yesh) [reply aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt]: Not sure if we have style guide, but I have updated the verb tense.

**Review by eeatonaws — APPROVED** [review]

**Pull request merged**